capsules/src/net/ipv6/ipv6.rs

```diff
@@ -301,6 +303,7 @@ impl IP6Header {
 /// This defines the currently supported `TransportHeader` types. The contents
 /// of each header is encapsulated by the enum type. Note that this definition
 /// of `TransportHeader`s means that recursive headers are not supported.
+#[derive(Copy, Clone)] // TODO: is this ok?
```

Comment:
Is this TODO still an open question? (fine if so, but doesn't compile without it -- if there's an underlying buffer / shallow-copy / etc concern behind this comment, might be nice to document that)

Reply:
Good catch, for now I think the overhead of this copy is fine, the largest header possible is a 40 byte TCP header which isn't huge and currently it looks like only 2 potentially unnecessary copies of a TransportHeader occur in the process of sending a packet, and changing to a design without copy would require a decent amount of restructuring.